presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewColumnMappingExtractor.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.sql.tree.ComparisonExpression;
+import com.facebook.presto.sql.tree.CreateMaterializedView;
+import com.facebook.presto.sql.tree.Query;
+import com.facebook.presto.sql.tree.SetOperation;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.MetadataUtil.toSchemaTableName;
+import static com.facebook.presto.spi.ConnectorMaterializedViewDefinition.TableColumn;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.NOT_SUPPORTED;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class MaterializedViewColumnMappingExtractor
+        extends MaterializedViewPlanValidator
+{
+    private final Analysis analysis;
+
+    private List<List<TableColumn>> mappedBaseColumns;
+
+    private MaterializedViewColumnMappingExtractor(Analysis analysis)
+    {
+        this.analysis = requireNonNull(analysis, "analysis is null");
+
+        this.mappedBaseColumns = new ArrayList<>();
+    }
+
+    public static List<List<TableColumn>> extractMappedBaseColumns(Analysis analysis)
+    {
+        checkState(
+                analysis.getStatement() instanceof CreateMaterializedView,
+                "Only support extracting of mapped columns from create materialized view query");
+
+        MaterializedViewColumnMappingExtractor extractor = new MaterializedViewColumnMappingExtractor(analysis);
+        Query viewQuery = ((CreateMaterializedView) analysis.getStatement()).getQuery();
+        extractor.process(viewQuery, new MaterializedViewPlanValidatorContext());
+
+        return ImmutableList.copyOf(extractor.mappedBaseColumns);
```

Comment:
- no need to copyOf if we are using ImmutableList.builder
- add a new private method called getMappedBaseColumns() to get mappedBaseColumns.